homeassistant/components/qwikswitch.py

```diff
@@ -120,18 +132,18 @@ def setup(hass, config):
             item[QS_TYPE] = 'switch'
             item[QS_NAME] = item[QS_NAME][:-7]
 
-    global QSUSB
-    if QSUSB is None:
-        QSUSB = {}
-    QSUSB[id(qsusb)] = qsusb
+    QSUSB[DOMAIN] = qsusb
 
     # Load sub-components for qwikswitch
     for comp_name in ('switch', 'light'):
         load_platform(hass, comp_name, 'qwikswitch',
-                      {'qsusb_id': id(qsusb)}, config)
+                      {'qsusb_id': DOMAIN}, config)
```

Comment:
If you don't have anything to pass in, just pass in an empty dict and make sure platforms check `discovery_info is not None`